### ISSUE TITLE
[💡 Feature]: allow in wdio-devtools-service coverageReporter multiple report types

### ISSUE THREAD

**Issue by bkrrrr** [webdriverio/webdriverio#8050]
### Is your feature request related to a problem?

Having multiple formats of coverage reports of the same test run is very common
  - HTML for test-writer
  - Text std as fast check for anyone
  - JSON for quality gates 

NYC can do this with multiple `---reporter` arguments [see](https://istanbul.js.org/docs/tutorials/mocha/)

But WDIO allows only one type and is not exposing the raw data, as NYC is doing it; in '.nyc_output'.

So there is no way to create another report.

### Describe the solution you'd like.

``` js
// wdio.conf.js
services: [
    ['devtools', {
        coverageReporter: {
            enable: true,
            type: [ 'html', 'text' ],  // => be able to declare this
            logDir: __dirname + '/coverage'
        }
    }]
]
```

### Describe alternatives you've considered.

Write the raw coverage results to .nyc_output. This would also enable to create multiple coverage reports. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by christian-bromann** [comment]
Thanks for reporting!

This sounds like a great addition to the service.

Any contributions that helps make this reality are highly appreciated. Please take a look into our [contribution guidelines](https://webdriver.io/docs/contribute) and let us know if you have any questions. Cheers!

**Comment by cenfun** [comment]
I created a WebdriverIO service for coverage reports, see https://github.com/cenfun/wdio-monocart-service

**Comment by christian-bromann** [comment]
Interesting 🤔 it seems like you are using the browser runner which offers istanbul based code coverage reporting as documented in [the docs](https://webdriver.io/docs/component-testing/coverage). This seems to be an interested approach to enable v8 coverage which might be a better alternative.

**Comment by cenfun** [comment]
@christian-bromann Yes, it generates reports directly from native v8 coverage data. Since both Nodejs and Chromium-based browsers Chrome/Edge are native support V8 coverage data, I personally prefer V8 because of its high performance and there is no need to instrument the source code with Istanbul. And we can also revert to using Istanbul by setting `coverageProvider` to `istanbul`.
```js
// wdio.conf.js
{
    services: [
        ['monocart', {
            name: 'My WebdriverIO Coverage Report',
            // v8 (default) or istanbul (requires instrumenting source code)
            // coverageProvider: 'v8',
            entryFilter: '**/src/**',
            reports: [
                'v8',
                'console-details'
            ],
            outputDir: './coverage-reports'
        }]
    ]
}
```

**Comment by christian-bromann** [comment]
@cenfun would you be interested submitting a PR to add this feature natively to the `@wdio/browser-runner`? I can see allowing users to pick between `v8` and `istanbul` (keeping istanbul as default in WebdriverIO v8 and choosing v8 as default for WebdriverIO v9 due to backwards compatibility.

**Comment by cenfun** [comment]
@christian-bromann Thank you for the invitation. I've done an initial study on project WebdriverIO, it might be a bit challenging for me personally to contribute. And I've found that it has two coverage services:
- 1, browser-runner with `coverage` option, seems it instrument the source code with `vite-plugin-istanbul`.
- 2, devtools-service with `coverageReporter` option, seems it can collect Istanbul data, I don't know its purpose.

The current `coverage` option is only for Istanbul, so add V8 coverage would definitely involve breaking changes.
I don't think it is applicable to WebdriverIO v8. It may need to be redesigned for WebdriverIO v9. 
My idea is:
- 1, for Istanbul
```
export const {
    // ...
    runner: ['browser', { 
      coverage: {
        enabled: true,
        provider: "istanbul",
        istanbulOptions: { /* previous coverage options, include both instrument and reporter options */}
      }
    }],
    // ...
}
```
- 1, for V8 coverage
```
export const {
    // ...
    runner: ['browser', { 
      coverage: {
        enabled: true,
        provider: "v8",
        v8Options: { /* v8 coverage options */}
      }
    }],
    // ...
}
```
I don't know if there is anything better, just for reference.

**Comment by christian-bromann** [comment]
> The current `coverage` option is only for Istanbul, so add V8 coverage would definitely involve breaking changes.

How different are the options? I wonder if we can keep them the same.

> * 2, devtools-service with `coverageReporter` option, seems it can collect Istanbul data, I don't know its purpose.

We have deprecated the devtools service and will remove its coverage functionality.

> it might be a bit challenging for me personally to contribute.

Happy to help on the way. We have [project office hours](https://webdriver.io/community/openofficehours) to help contributor on questions contributing to the project.

**Comment by cenfun** [comment]
> How different are the options? I wonder if we can keep them the same.
Almost different. For Istanbul, it requires to instrument the source code but no need for V8. For V8 coverage, it requires to filter the results for coverage data list but not need for Istanbul, see [here](https://github.com/cenfun/monocart-coverage-reports?#collecting-istanbul-coverage-data)

For [MCR](https://github.com/cenfun/monocart-coverage-reports), it actually doesn't support instrumenting source code, the support is more for V8 coverage.

**Comment by christian-bromann** [comment]
If user facing interfaces don't change we could move forward with this in the following way: introduce a `provider` option and keep `istanbul` as default for v8 and make it `v8` for `v9` 😆 , wdyt?
